Library/Homebrew/dev-cmd/bump-formula-pr.rb

```diff
@@ -314,73 +314,39 @@ def bump_formula_pr
       end
     end
 
-    unless Formula["hub"].any_version_installed?
-      if ARGV.dry_run?
-        ohai "brew install hub"
-      else
-        safe_system "brew", "install", "hub"
-      end
-    end
-
     formula.path.parent.cd do
       branch = "#{formula.name}-#{new_formula_version}"
       git_dir = Utils.popen_read("git rev-parse --git-dir").chomp
       shallow = !git_dir.empty? && File.exist?("#{git_dir}/shallow")
 
-      hub_args = []
-      git_final_checkout_args = []
-      if ARGV.include?("--no-browse")
-        git_final_checkout_args << "--quiet"
-      else
-        hub_args << "--browse"
-      end
-
       if ARGV.dry_run?
-        ohai "hub fork # read $HUB_REMOTE"
+        ohai "fork"
         ohai "git fetch --unshallow origin" if shallow
         ohai "git checkout --no-track -b #{branch} origin/master"
         ohai "git commit --no-edit --verbose --message='#{formula.name} #{new_formula_version}#{devel_message}' -- #{formula.path}"
         ohai "git push --set-upstream $HUB_REMOTE #{branch}:#{branch}"
-        ohai "hub pull-request #{hub_args.join(" ")} -m '#{formula.name} #{new_formula_version}#{devel_message}'"
+        ohai "create pull-request with GitHub API"
         ohai "git checkout -"
       else
-        reply = IO.popen(["hub", "fork"], "r+", err: "/dev/null") do |io|
-          reader = Thread.new { io.read }
-          sleep 1
-          io.close_write
-          reader.value
-        end
 
-        if reply.to_s.include? "username:"
+        begin
+          response = GitHub.create_fork(formula.tap.full_name)
+        rescue
           formula.path.atomic_write(backup_file) unless ARGV.dry_run?
-          git_path = "$(brew --repo #{formula.tap})" if formula.tap
-          git_path ||= formula.path.parent
-          odie <<~EOS
-            Retry after configuring hub by running:
-              hub -C "#{git_path}" fork
-            Or setting HOMEBREW_GITHUB_TOKEN with at least 'public_repo' scope.
-          EOS
+          odie "Unable to fork"
         end
 
-        remote = reply[/remote:? (\S+)/, 1]
-
-        # repeat for hub 2.2 backwards compatibility:
-        remote = Utils.popen_read("hub", "fork", err: :out)[/remote:? (\S+)/, 1] if remote.to_s.empty?
-
-        if remote.to_s.empty?
-          formula.path.atomic_write(backup_file) unless ARGV.dry_run?
-          odie "cannot get remote from 'hub'!"
-        end
+        remote_url = response["clone_url"]
```

Comment:
Can we use `#fetch` instead of `#[]` throughout? It means that if a key isn’t present, we’ll get an exception right away, rather than a hard to debug `nil` error.

Reply:
Addressed.